code/modules/clothing/clothing.dm

```diff
@@ -198,6 +200,28 @@ BLIND     // can't see anything
 		var/mob/M = loc
 		M.update_inv_shoes()
 
+/obj/item/clothing/shoes/attackby(obj/item/I, mob/user, params)
+	if(!can_hold_knives)
+		return
+	if(held_knife)
+		user << "<span class='notice'>There's already something in [src].</span>"
+		return
+	if(istype(I, /obj/item/weapon/kitchen/knife) || istype(I, /obj/item/weapon/pen) || istype(I, /obj/item/weapon/switchblade))//can hold both regular pens and energy daggers. made for your every-day tactical librarians/murderers.
+		if(!user.drop_item())
+			return
+		I.loc = src
+		held_knife = I
+		user << "<span class='notice'>You discreetly slip [I] into [src]. Alt-click [src] to remove it.</span>"

```

Comment:
It would be nice if this proc could call parent somewhere, for future-proofing. (Also would be nice if you did the same for banana_shoes attackby() if possible)